### PR TITLE
feat: close sliding panel on escape key, issue #8225

### LINKED ISSUE #8225: Escape key should close sliding window in UI
# Summary

In the UI, when a slider window opens, it should be closable using the Escape key.

# Motivation

Better user experience for people using the UI with keyboard and mouse in combination. Being able to close popups/modals using the Escape key seems best practice.

# Proposal

Have the slider closable using the Escape key.

### PR DESCRIPTION
**Pull request by reginaelyse**
Closes #8225
NOTE: need to merge https://github.com/argoproj/argo-ui/pull/184 first 

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Review by rbreeze — APPROVED** [review]
LGTM once Argo UI PR is merged

**Review by saumeya — APPROVED** [review]

**Pull request merged**
